pyomo/contrib/doe/examples/reactor_compute_FIM.py

```diff
@@ -106,6 +110,24 @@ def main():
     relative_error = abs(np.log10(result.det) - 2.99)
     assert relative_error < 0.01
 
+    ### Compute the FIM of a square model-based Design of Experiments problem
+    ### Test if we can set up design variables as Param instead of Var
+    doe_object2 = DesignOfExperiments(
+        parameter_dict,  # parameter dictionary
+        exp_design,  # DesignVariables object
+        measurements,  # MeasurementVariables object
+        create_model_design_as_param,  # create model function
+        discretize_model=disc_for_measure,  # discretize model function
+    )
+
+    result2 = doe_object2.compute_FIM(
+        mode="direct_kaug",  # calculation mode
+        scale_nominal_param_value=True,  # scale nominal parameter value
+        formula="central",  # formula for finite difference
+    )
+
+    result2.result_analysis()
```

Comment:
Are you expecting the results to be the same as the previous case? If so I think you should have the same assert statements checking the trace and determinant.

Reply:
I modified this test to use both `Param` components for parameters and variables, change one parameter value, use assertions to check if the generated model uses the updated model parameter value, and check the trace and determinant.